Src/BootCamp.Chapter1/ArrayOperations.cs

```diff
@@ -75,7 +146,21 @@ public static int[] InsertFirst(int[] array, int number)
         /// <returns>A new array with element added in the end of array. If an array is empty or null, returns new array with number in it.</returns>
         public static int[] InsertLast(int[] array, int number)
         {
-            // ToDo: implement.
+            if (array == null)
```

Comment:
Reuse the functions

Reply:
Thanks! Could I get a bit more help on how I can minimize repetition? I've made a function called ValidateArrayForManipulation(int[] array), but I think I'm getting caught on the differences between null and [0]. Some of the code (that remains) seems to need validation in a certain order.

Reply:
InsertFirst(int[] array, int element) => InsertElementAt(array, element, 0);

Reply:
Focus on the big chunks. 
InsertLast = InsertAt last index position
InsertFirst = InsertAt first index position
....

Reply:
I did some more, looking for any redundancy. My code looks messy, with a combination of "primary" functions to manage the work and "helper" functions to remove duplication (and get more descriptive of what it's doing). Should I also be organizing them together, at the bottom of the file? Better yet, do I put them in a different file?

Reply:
This is a good attempt, but the contents of insert and remove functions still overlap a lot. WHat I meant was:
```
int[] InsertLast(int[] array, int element)
{
  return InsertAt(array, element, array.Length-1);
}
```

Reply:
thanks. I think I go a lot closer. the related ...First and ...Last functions now call the ...At function with the specified parameters.